app/src/main/java/com/babylon/wallet/android/presentation/transaction/TransactionReviewScreen.kt

```diff
@@ -182,19 +174,49 @@ private fun TransactionPreviewContent(
     onFeePaddingAmountChanged: (String) -> Unit,
     onTipPercentageChanged: (String) -> Unit,
     onViewDefaultModeClick: () -> Unit,
-    onViewAdvancedModeClick: () -> Unit
+    onViewAdvancedModeClick: () -> Unit,
+    dismissNoMnemonicErrorDialog: () -> Unit,
+    dismissDepositRulesErrorDialog: () -> Unit
 ) {
     val modalBottomSheetState = rememberModalBottomSheetState(
         initialValue = ModalBottomSheetValue.Hidden,
         skipHalfExpanded = true
     )
     val snackBarHostState = remember { SnackbarHostState() }
 
-    SnackbarUIMessage(
-        message = state.error,
-        snackbarHostState = snackBarHostState,
-        onMessageShown = onMessageShown
-    )
+    state.error?.let { transactionError ->
+        if (transactionError.isPreviewedInDialog) {
+            if (transactionError.isNoMnemonicErrorVisible) {
+                BasicPromptAlertDialog(
+                    finish = {
+                        dismissNoMnemonicErrorDialog()
+                    },
+                    title = stringResource(id = R.string.transactionReview_noMnemonicError_title),
+                    text = stringResource(id = R.string.transactionReview_noMnemonicError_text),
+                    dismissText = null
+                )
+            }
+            if (transactionError.isDepositRulesErrorVisible) {
+                val body = stringResource(id = R.string.error_transactionFailure_reviewFailure)
+                val subBody = stringResource(id = R.string.error_transactionFailure_doesNotAllowThirdPartyDeposits)
+                BasicPromptAlertDialog(
+                    finish = {
+                        dismissDepositRulesErrorDialog()
+                    },
+                    title = stringResource(id = R.string.common_errorAlertTitle),
+                    text = "$body\n\n$subBody",
+                    confirmText = stringResource(id = R.string.common_ok),
+                    dismissText = null
+                )
+            }
+        } else {
+            SnackbarUIMessage(
+                message = transactionError,
+                snackbarHostState = snackBarHostState,
+                onMessageShown = onMessageShown
+            )
+        }
```

Comment:
It is almost to what I described. Again this will end up defining a lot of if else statements for different errors. Isn't the point of `toUserFriendlyMessage()` to return us the correct text?
```
state.error?.let { transactionError ->
        if (transactionError.isPreviewedInDialog) {
                BasicPromptAlertDialog(
                    finish = {
                        dismissNoMnemonicErrorDialog()
                    },
                    title = transactionError.title(), // This may need some attention to create custom titles or leave a generic one.
                    text = stringResource(transactionError.toUserFriendlyMessage()),
                    dismissText = null
                )
        } else {
            SnackbarUIMessage(
                message = transactionError,
                snackbarHostState = snackBarHostState,
                onMessageShown = onMessageShown
            )
        }
```

Reply:
@micbakos-rdx Fixed.